src/jit/rangecheck.cpp

```diff
@@ -129,12 +129,6 @@ bool RangeCheck::BetweenBounds(Range& range, int lower, GenTreePtr upper)
             return false;
         }
 
-        // If lower limit is len return false.
-        if (range.LowerLimit().IsArray())
```

Comment:
This one was probably bogus. In the `if` statement below any constant lower limit that's positive results in range check elimination. But an array length lower limit does not, even though array lengths are always positive. Hmm.

Anyway, this change doesn't alter this behavior. After this change we'll end up with a `keBinOpArray` with a 0 constant and that one too is rejected in another `if` statement below this one.